engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1130,6 +1138,128 @@ namespace dmGameSystem
+     *
+     * @name physics.set_group
+     * @param url [type:string|hash|url] the collision object affected.
+     * @param group [type:string] the new group name to be assigned.
+     * ```lua
+     * function example()
+     *      physics.set_group("#collisionobject", "enemy")
+     * end
+     * ```
+     */
+    static int Physics_SetGroup(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        dmhash_t group_id = dmScript::CheckHashOrString(L, 2);
+        
+        dmGameSystem::SetCollisionGroup(comp_world, comp, group_id);
+
+        return 0;
+    }    
+
+    /*# returns the group of a collision object
+     *
+     * Returns the group name of a collision object as a hash.
+     *
+     * @name physics.get_group
+     * @param url [type:string|hash|url] the collision object to return the group of.
+     * @return [type:hash] hash value of the group 
+     * ```lua
+     * function example()
+     *     local grp = physics.get_group("#collisionobject")
+     *     assert( grp == hash("enemy") )
+     * end
+     * ```
+     */    
+    static int Physics_GetGroup(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        
+        dmhash_t group_hash = dmGameSystem::GetCollisionGroup(comp_world, comp);
+        dmScript::PushHash(L, group_hash);
+        
+        return 1;
+    }    
+
+    /*# updates the mask of a collision object
+     *
+     * Sets or clears the masking of a group (maskbit) in a collision object.
+     *
+     * @name physics.set_maskbit
+     * @param url [type:string|hash|url] the collision object to change the mask of.
+     * @param group [type:string] the name of the group (maskbit) to modify in the mask.
+     * @param [type:boolean] boolean value of the new maskbit. 'true' to enable, 'false' to disable.
+     * ```lua
+     * function example()
+     *     -- no longer collide with the "enemy" group
+     *     physics.set_maskbit("#collisionobject","user",enemy)
+     * end
+     * ```
+     */
+    static int Physics_SetMaskBit(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        dmhash_t group_id = dmScript::CheckHashOrString(L, 2);
+        bool boolvalue = CheckBoolean(L, 3);
+        
+        dmGameSystem::SetCollisionMaskBit(comp_world, comp, group_id, boolvalue);
```

Comment:
Error handling?

Reply:
There is already a warning written to log (in GetGroupBitIndex()) if the mask does not exist.

What's the proper way to handle this ?

Reply:
I think given our discussion above, we've now decided that these errors should produce a `luaL_error` call.